lib/model/internal_link.dart

```diff
@@ -203,6 +203,7 @@ Narrow? _interpretNarrowSegments(List<String> segments, PerAccountStore store) {
       case IsOperand.private:
       case IsOperand.alerted:
       case IsOperand.starred:
+        return const StarredMessagesNarrow();
```

Comment:
```
internal_link: Parse "/is/starred" internal links.
```

This is a bug. :) Since cases before `IsOperand.starred` are empty, they'll also give `const StarredMessagesNarrow()`, just like the `IsOperand.starred` case.

https://dart.dev/language/branches#switch-statements

> Non-empty `case` clauses jump to the end of the switch after completion. They do not require a `break` statement.

It's a common pattern to want multiple cases to give the same thing, and to express that with empty `case` clauses, as here.

To fix (here and in the test, which has the same bug), I would just reorder:

```dart
    switch (isElementOperands.single) {
      case IsOperand.mentioned:
        return const MentionsNarrow();
      case IsOperand.starred:
        return const StarredMessagesNarrow();
      case IsOperand.dm:
      case IsOperand.private:
      case IsOperand.alerted:
      case IsOperand.followed:
      case IsOperand.resolved:
      case IsOperand.unread:
      case IsOperand.unknown:
        return null;
    }
```